fms/models/gpt_bigcode.py

```diff
@@ -58,20 +58,19 @@ def __init__(self, config: GPTBigCodeConfig):
 
     def forward(
         self,
-        x: torch.LongTensor,
+        x: torch.Tensor,
         *,
         mask: Optional[torch.Tensor] = None,
-        position_ids: Optional[torch.LongTensor] = None,
+        position_ids: Optional[torch.Tensor] = None,
```

Comment:
Should this not just be a long tensor?

Reply:
For LongTensor/FloatTensor/BoolTensor, these aren't actually used in pytorch type annotations anywhere and they don't inherit from torch.Tensor, so are effectively unsupported. I.e. if you have `t:BoolTensor = x == y`, you're going to get a failure from mypy because the `==` returns a Tensor not a BoolTensor.

I also seem to remember these types being discouraged from use when I was working on PyTorch internals. I think they were only kept for BC reasons but aren't generally recommended for use.